### ISSUE TITLE
Is importing an empty file supported?

### ISSUE THREAD

**Issue by MeoMix** [systemjs/systemjs#762]
Hey,

I encountered a rather nasty bug this evening. I attempted to import a file which had only commented out lines within it. Attempting to do so caused my browser to become unresponsive. Looking at the 'Network' tab in Chrome Dev Tools I can see that the last request to successfully complete is the file in question. All XHRs after that point remain pending.

Adding anything of substance to the file and everything went back to working.

Chrome Version 45.0.2454.85 m.

Is this a known issue? Weird edge case? Something corrupt on my end?

**Comment by guybedford** [comment]
Thanks for posting. Can you share an isolated replication or steps to replicate for this?

**Comment by MeoMix** [comment]
https://github.com/MeoMix/jspmdebug

I built a repo with an example inside of it.

Check it out, run the standard `npm install jspm` `jspm init` and accept all defaults.

I was testing this in a Chrome extension environment, so for me I would then:

- go to chrome://extensions
- check 'developer mode' checkbox
- check 'load unpacked extension' and select parent directory of manifest.json
- navigate to chrome-extension://jbnkffmindojffecdhbbmekbmkkfpmjd/test.html

Observe that page is totally unresponsive. If you open developer tools before navigating to that link then you can see all the network requests go and then it just hangs after trying to load `test.js`

If I delete all the commented out lines (or even just some of them!) from `test.js` then everything runs as expected.

**Comment by guybedford** [comment]
Thanks for posting this.

I've managed to trace it down that the following regular expression will take a very long time:

```javascript
var a = "//// /collection/\n//import 'test/background/collection/clientErrors.spec';\n//import 'test/background/collection/playlistItems.spec';\n//import 'test/background/collection/playlists.spec';\n//import 'test/background/collection/searchResults.spec';\n//import 'test/background/collection/videos.spec';\n//import 'test/background/collection/streamItems.spec';\n\n//// /model/\n//import 'test/background/model/activePlaylistManager.spec';\n//import 'test/background/model/clientErrorManager.spec';\n//import 'test/background/model/dataSource.spec';\n//import 'test/background/model/playlistItem.spec';\n//import 'test/background/model/playlistItems.spec';\n//import 'test/background/model/relatedVideosManager.spec';\n//import 'test/background/model/signInManager.spec';\n//import 'test/background/model/user.spec';\n//import 'test/background/model/youTubeV3API.spec';"

a.match(/^\s*(\s*\/\/[^\n]*|\s*"[^"]+"\s*;?|\s*'[^']+'\s*;?|\/\*[^\*]*(\*(?!\/)[^\*]*)*\*\/)*\s*System\.register(Dynamic)?\s*\(/)
```

I didn't think there was any bad backtracking going on here, but apparently there must be.

Will have to debug this further to correct the regex.

**Comment by guybedford** [comment]
NB this should be ported for 0.18 before the 0.19 release.

**Comment by guybedford** [comment]
Released in 0.19.0.
